src/uk/gov/defra/ffc/Provision.groovy

```diff
@@ -25,6 +27,95 @@ class Provision implements Serializable {
+          ctx.sh("docker-compose -p $repoName-$pr -f docker-compose.migrate.yaml run schema-up")
+        }
+        ctx.sh("docker-compose -p $repoName-$pr -f docker-compose.migrate.yaml run --no-deps database-up")
+      }
+    }
+  }
+
+  private static def deletePrDatabase(ctx, environment, repoName, pr) {
+    if (pr != '' && ctx.fileExists('./docker-compose.migrate.yaml')) {
+      def migrationFolder = 'migrations'
+      getMigrationFiles(ctx, migrationFolder)
+      
+      ctx.withEnv(getMigrationEnvVars(ctx, environment, repoName, pr)) {
+        ctx.dir(migrationFolder) {
+           // removing the schema removes the database migrations within that schema, so is unneccessary
+           ctx.sh("docker-compose -p $repoName-$pr -f docker-compose.migrate.yaml run --no-deps schema-down")
+        }
+      }
+    }
+  }
+
+  private static def getMigrationFiles(ctx, destinationFolder){
+    def resourcePath = 'uk/gov/defra/ffc/migration'
+    ctx.sh(" rm -rf $destinationFolder")
+    getResourceScript(ctx, "$resourcePath/scripts", 'schema-up', "$destinationFolder/scripts")
+    getResourceScript(ctx, "$resourcePath/scripts", 'schema-down', "$destinationFolder/scripts")
+    getResourceFile(ctx, resourcePath, 'docker-compose.migrate.yaml', destinationFolder)
+    getResourceFile(ctx, "$resourcePath/changelog", 'schema.changelog.xml', "$destinationFolder/changelog")
+  }
+
+  private static def getResourceFile(ctx, resourcePath, filename, destinationFolder, makeExecutable = false){
+    def fileContent = ctx.libraryResource("$resourcePath/$filename")
+    ctx.writeFile(file: "$destinationFolder/$filename", text: fileContent, encoding: "UTF-8")
+    if (makeExecutable) {
+      ctx.sh("chmod 777 ./$destinationFolder/$filename")
+    }
+    ctx.echo "written $filename to $destinationFolder"
+  }
+
+  private static def getResourceScript(ctx, resourcePath, filename, destinationFolder){
+    getResourceFile(ctx, resourcePath, filename, destinationFolder, true)
+  }
+
+  private static def getMigrationEnvVars(ctx, environment, repoName, pr) {
+    def searchKeys = [
+      'POSTGRES_HOST',
+      'POSTGRES_ADMIN_USERNAME',
+      'POSTGRES_ADMIN_PASSWORD',
+      'POSTGRES_SCHEMA_PASSWORD'
+    ]
+    def appConfigPrefix = environment + '/'
+    def appConfigValues = Utils.getConfigValues(ctx, searchKeys, appConfigPrefix, Utils.defaultNullLabel, false)
+    appConfigValues['POSTGRES_ADMIN_PASSWORD'] = escapeQuotes(appConfigValues['POSTGRES_ADMIN_PASSWORD'])
+    appConfigValues['POSTGRES_SCHEMA_PASSWORD'] = escapeQuotes(appConfigValues['POSTGRES_SCHEMA_PASSWORD'])
+
+    def migrationEnvVars = appConfigValues.collect { "$it.key=$it.value" }
+
+    def schemaName = repoName.replace('-','_') + pr
+    def schemaRole = repoName.replace('-','_') + pr + "role"
+    def schemaUser = getSchemaUserWithHostname(schemaRole, appConfigValues['POSTGRES_HOST'])
+    def databaseName = repoName.replace('-','_').replace('_service', '')
```

Comment:
the database name doesn't contain the PR number, as it is the database in postgres where the schema is created.
In this case I think it's a little clearer to replace a simple string on the `repoName`, rather than replace the role and pr on a previously replaced string, so will leave as it is.